pkg/migrations/sql/idempotent/base.sql

```diff
@@ -610,6 +610,23 @@ $$
 LANGUAGE SQL STABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION SCHEMA_PROM.label_key_position(text, text) to prom_reader;
 
+-- public function to delete a metric.
+CREATE OR REPLACE FUNCTION prom_api.delete_metric(metric_name_to_be_deleted text) RETURNS VOID
```

Comment:
You also have to drop the series and metric views create by `create_series_view` and `create_metric_view` functions. also rows in label_key_position table. Finally we should probably clean up any unused labels - see https://github.com/timescale/promscale/blob/master/pkg/migrations/sql/idempotent/base.sql#L1286.

Reply:
Won't the dropping of views would be handled by `CASCADE`?